File: frontend/src/game/ai/AIStrategy.ts
import type { BoardState } from '../engine/types';
import type { GameRules } from '../engine/rules/GameRules';

export interface AIStrategy {
  readonly name: string;

  getBestMove(board: BoardState, aiPlayer: number, rules: GameRules, options?: AIOptions): [number, number] | null;
}

export interface AIOptions {
  depth?: number;
  [key: string]: any;
}

File: backend/src/main/java/com/gobang/ai/AIStrategy.java
package com.gobang.ai;

import com.gobang.engine.rules.GameRules;

public interface AIStrategy {
    String getName();

    int[] getBestMove(int[][] board, int aiPlayer, GameRules rules, AIOptions options);

    class AIOptions {
        private int depth = 3;

        public AIOptions() {}

        public AIOptions(int depth) {
            this.depth = depth;
        }

        public int getDepth() {
            return depth;
        }

        public void setDepth(int depth) {
            this.depth = depth;
        }
    }
}
